### PR TITLE
(SIMP-1695) Use simp_options module for global catalysts

### PR DESCRIPTION
**Pull request by lnemsick-simp**
SIMP-2196 #close
SIMP-1695 #comment

### CODE REVIEW

**Review by jeefberkey — APPROVED** [review]

**Review by nick-markowski — CHANGES_REQUESTED** [review]
This module needs to require the simp_opt module (metadata, build/requires).  I believe it also needs a major bump.

EDIT: Apparently not. Ticket updated.

**Review by nick-markowski — APPROVED** [review]

File: manifests/utils.pp [hunk 1]
```diff
@@ -13,8 +13,8 @@
 #
 class snmpd::utils (
   $rsync_source  = "snmp_${::environment}/mibs",
-  $rsync_server  = hiera('rsync::server'),
-  $rsync_timeout = hiera('rsync::timeout','2')
+  $rsync_server  = lookup('::simp_options::rsync::server',  { 'default_value' => '127.0.0.1', 'value_type' => String}),
+  $rsync_timeout = lookup('::simp_options::rsync::timeout', { 'default_value' => '2', 'value_type' => String }),
```
Comment (lamawithonel) [inline manifests/utils.pp]: These should be static. The `lookup()` function does not look at the variable namespace, so if a user modifies `::simp_options` via profile and sets a different value in Hiera the values will diverge. The logic should go in the module's `<module_name>::data` function.

See: https://docs.puppet.com/puppet/latest/lookup_quick_module.html#more-about-the-function-data-provider

**Comment by lnemsick-simp** [comment]
lookup should be lookup('simp_options::rsync::server',  { 'default_value' => '127.0.0.1', 'value_type' => String}),

Don't merge

**Comment by lnemsick-simp** [comment]
Work incomplete. Don't approve!!!

**Comment by lnemsick-simp** [comment]
Ready for re-review

**Review by nick-markowski — CHANGES_REQUESTED** [review]
This looks great, but we should include simp_options in fixtures for best testing

**Review by nick-markowski — APPROVED** [review]

**Pull request merged**